pkg/util/validation/limits.go

```diff
@@ -975,19 +977,26 @@ func newLimitsWithExtensions(limits *plainLimits) (any interface{}, getExtension
 
 	// typ is the type of the new struct.
 	typ := reflect.StructOf(fields)
-
 	// cfg is an instance of a pointer to a new struct.
-	cfg := reflect.New(typ).Interface()
+	cfg := reflect.New(typ)
+
+	// Set default values of each field
+	// In other words:
+	//     cfg.EXTNAME1 = cfg.EXTNAME1.Default()
+	for i := range limitsExtensionsFields {
+		res := cfg.Elem().Field(i).MethodByName("Default").Call(nil)
```

Comment:
[nit] Instead of relying on `MethodByName` we could store a "defaults getter" function in `MustRegisterExtension`, since we have access to it there.

Reply:
Good idea, done in https://github.com/grafana/mimir/pull/4418/commits/2b6305fa55f2806229a38dd0cff681bc10bde0a6